tests/integration/corrupt-dump.tcl

```diff
@@ -44,9 +44,10 @@ test {corrupt payload: #7445 - without sanitize - 2} {
 test {corrupt payload: hash with valid zip list header, invalid entry len} {
     start_server [list overrides [list loglevel verbose use-exit-on-panic yes crash-memcheck-enabled no] ] {
         r config set sanitize-dump-payload no
-        r restore key 0 "\x0D\x1B\x1B\x00\x00\x00\x16\x00\x00\x00\x04\x00\x00\x02\x61\x00\x04\x02\x62\x00\x04\x14\x63\x00\x04\x02\x64\x00\xFF\x09\x00\xD9\x10\x54\x92\x15\xF5\x5F\x52"
-        r config set hash-max-ziplist-entries 1
-        catch {r hset key b b}
+        # cause an assertion when converting ziplist to listpack
+        catch {
+            r restore key 0 "\x0D\x1B\x1B\x00\x00\x00\x16\x00\x00\x00\x04\x00\x00\x02\x61\x00\x04\x02\x62\x00\x04\x14\x63\x00\x04\x02\x64\x00\xFF\x09\x00\xD9\x10\x54\x92\x15\xF5\x5F\x52"
+        }
```

Comment:
few issues here:
first, the obvious one is that we ignore the error returned from `catch` and the test doesn't really cares if the RESTORE failed or succeeded.
this would have been ok, if we had an assertion after it that makes sure the server died on assertion.
i.e.
```
        assert_equal [count_log_message 0 "crashed by signal"] 0
        assert_equal [count_log_message 0 "ASSERTION FAILED"] 1
```
but instead we have some (outdated?) verification that makes sure there was no ziplist assertion.

secondly, as i suggested in my review of rdb.c, maybe we wanna make the deep validation of ziplists unconditional, since we're anyway doing an O(n) iteration on the members, so O(2n) is not that bad.
if we do that, we would be expecting a `assert_match "*Bad data format*" $err` response from RESTORE.